### PR TITLE
Bib update

### PR DESCRIPTION
**Pull request by wence-**
Update some more of the references and add a few new ones.

### CODE REVIEW

File: docs/source/_static/bibliography.bib [hunk 1]
```diff
@@ -33,6 +33,16 @@ @Article{Homolya2016
   url =          {http://arxiv.org/abs/1505.03357}
 }
 
+@Misc{Homolya2017,
```
Comment (dorugeber) [inline docs/source/_static/bibliography.bib]: why not Article?

Comment (dorugeber) [inline docs/source/_static/bibliography.bib]: nvm

File: docs/source/_static/firedrake-apps.bib [hunk 2]
```diff
@@ -1,16 +1,39 @@
 @Article{Chang2017,
```
Comment (dorugeber) [inline docs/source/_static/firedrake-apps.bib]: Probably should be Misc, judging by other changes?

File: docs/source/_static/firedrake-apps.bib [hunk 3]
```diff
@@ -46,26 +69,18 @@ @InProceedings{Bokhove2016
...
+@Article{Braun2017,
+  author =       {Moritz Braun and Kingsley O. Obodo},
+  title =        {Multi-domain muffin tin finite element density
+                  functional calculations for small molecules},
+  journal =      {Computers & Mathematics with Applications},
+  year =         2017,
+  issn =         {0898-1221},
+  doi =          {10.1016/j.camwa.2016.12.003},
+  url =
+                  {http://www.sciencedirect.com/science/article/pii/S0898122116306745},
```
Comment (miklos1) [inline docs/source/_static/firedrake-apps.bib]: Looks like a spurious newline.

**Pull request merged**